### PR TITLE
doc updates for yaml

### PR DESCRIPTION
**Pull request by TorreZuk**
* adds basic documentation

### CODE REVIEW

**Review by amcamd — APPROVED** [review]
LGTM

File: docs/source/clients.rst [hunk 1]
```diff
@@ -59,14 +59,48 @@ Logging affects performance, so only use it to log the command to copy and chang
 
 Note that hipblas-bench also has the flag ``-v 1`` for correctness checks.
 
+If multiple arguments or even multiple functions need to be benchmarked there is support for data driven benchmarks via a yaml format specification file.
+
+.. code-block:: bash
+
+   ./hipblas-bench --yaml <file>.yaml
+
+An example yaml file that is used for a smoke test is hipblas_smoke.yaml but other examples can be found in the rocBLAS repositiory.
```
Comment (NaveenElumalaiAMD) [inline docs/source/clients.rst:68]: repositiory -> repository

File: docs/source/clients.rst [hunk 2]
```diff
@@ -59,14 +59,48 @@ Logging affects performance, so only use it to log the command to copy and chang
...
+
+   --gtest_filter=POSTIVE_PATTERNS[-NEGATIVE_PATTERNS]
+
+If specific function arguments or even multiple functions need to be tested there is support for data driven testing via a yaml format test specification file.
+
+.. code-block:: bash
+
+   ./hipblas-test --yaml <file>.yaml
+
+An example yaml file that is used to define a smoke test is hipblas_smoke.yaml but other examples can be found in the rocBLAS repositiory.  Yaml based
```
Comment (NaveenElumalaiAMD) [inline docs/source/clients.rst:100]: repositiory -> repository

**Review by NaveenElumalaiAMD — APPROVED** [review]
LGTM! minor spelling changes.

**Pull request merged**